src/Public/Responses.ps1

```diff
@@ -830,30 +908,48 @@ function Write-PodeYamlResponse {
         $StatusCode = 200
     )
 
-    switch ($PSCmdlet.ParameterSetName.ToLowerInvariant()) {
-        'file' {
-            if (Test-PodePath $Path) {
-                $Value = Get-PodeFileContent -Path $Path
-            }
+    begin {
+        $pipelineValue = @()
+    }
+
+    process {
+        if ($PSCmdlet.ParameterSetName -eq 'Value') {
+            $pipelineValue += $_
         }
+    }
 
-        'value' {
-            if ($Value -isnot [string]) {
-                if ( $Depth -gt 0) {
-                    $Value = ConvertTo-PodeYaml -InputObject $Value -Depth $Depth
+    end {
+
+        switch ($PSCmdlet.ParameterSetName.ToLowerInvariant()) {
+            'file' {
+                if (Test-PodePath $Path) {
+                    $Value = Get-PodeFileContent -Path $Path
                 }
-                else {
-                    $Value = ConvertTo-PodeYaml -InputObject $Value
+            }
+
+            'value' {
+                if ($pipelineValue) {
+                    $Value = $pipelineValue
+                }
+
+                if ($Value -is [hashtable]) {
+                    $Value = @(foreach ($v in $Value) {
+                            New-Object psobject -Property $v
+                        })
+                }
```

Comment:
Is this needed for YAML? Because `ConvertTo-PodeYamlInternal` handles hashtables 🤔

Reply:
no yaml doesn't need it like json.
only CSV and XML need it

Reply:
this was a refuse from my tests.
it's useless and it's not required